File: src/main/java/nieboczek/lifestolen/serializer/base/IdentifierSerializer.java
package nieboczek.lifestolen.serializer.base;

import nieboczek.lifestolen.serializer.lang.SerializedStringBuilder;
import nieboczek.lifestolen.serializer.lang.TokenStream;
import nieboczek.lifestolen.serializer.lang.TokenType;

public class IdentifierSerializer extends Serializer<String> {
    private IdentifierSerializer() {}

    public static IdentifierSerializer of() {
        return new IdentifierSerializer();
    }

    @Override
    public void serialize(String value, SerializedStringBuilder builder) {
        builder.text(value);
    }

    @Override
    public String deserialize(TokenStream stream) {
        return stream.nextTokenText(TokenType.IDENTIFIER);
    }
}

File: src/main/kotlin/nieboczek/lifestolen/serializer/base/IdentifierSerializer.kt
package nieboczek.lifestolen.serializer.base

import nieboczek.lifestolen.serializer.lang.SerializedStringBuilder
import nieboczek.lifestolen.serializer.lang.TokenStream
import nieboczek.lifestolen.serializer.lang.TokenType

class IdentifierSerializer : Serializer<String>() {
    override fun serialize(value: String, builder: SerializedStringBuilder) {
        builder.text(value)
    }

    override fun deserialize(stream: TokenStream): String {
        return stream.nextTokenText(TokenType.IDENTIFIER)
    }
}
